### PR TITLE
Subprocess: Respect LXD_SECURITY_APPARMOR env var

### LINKED ISSUE #11675: LXD_SECURITY_APPARMOR=false broken
# Required information

 * Distribution: Debian
 * Distribution version: bullseye
 * The output of "lxc info" or if that fails:
   * Kernel version: `uname -rv` `5.15.0-25-generic #25-Ubuntu SMP Wed Mar 30 15:54:22 UTC 2022`
   * LXC version: `lxc --version` `5.13` built from master 3 weeks ago
   * LXD version: `lxd --version` `5.13` built from master 3 weeks ago 
   * Storage backend in use: `dir`

# Issue description

I'm attempting to run lxd in the following configuration:
- host: Ubuntu, using snapd
  - ctn: Ubuntu, using snapd
    - ctn: ??

Containers run with nesting enabled and the containers unprivileged.

AppArmor prevents running various pieces of lxd and the `LXD_SECURITY_APPARMOR` envvar is not working as expected.

# Steps to reproduce

Running `snapd` in the inner container fails due to AppArmor so I attempted my build at https://github.com/antifob/lxd-ci/
AppArmor then caused a failure to run `dnsmasq`:

```
root@ctn:~# lxc network create lxdbr0
Error: The DNS and DHCP service exited prematurely: Process exited with non-zero value 1 ("aa-exec: ERROR: profile 'lxd_dnsmasq-lxdbr0_</var/lib/lxd>' does not exist")
```

So I set `LXD_SECURITY_APPARMOR=false` according to https://linuxcontainers.org/lxd/docs/latest/environment/
Same error regarding `dnsmasq` so I hacked around it by setting `raw.dnsmasq`.

```
root@ctn:~# lxc network create lxdbr0 raw.dnsmasq='#'
Network lxdbr0 created
```

Then, I cannot launch a container because `tar` runs with an AppArmor profile:
```
root@ctn:~# lxc launch images:alpine/edge z
Creating z
Error: Failed instance creation: Failed creating instance from image: Unpack failed: Failed to run: tar --wildcards --exclude=dev/* --exclude=./dev/* --exclude=rootfs/dev/* --exclude=rootfs/./dev/* --restrict --force-local -C /var/lib/lxd/storage-pools/default/containers/z --numeric-owner --xattrs-include=* -Jxf -: Process exited with non-zero value 1 (aa-exec: ERROR: profile 'lxd_archive-var-lib-lxd-storage-pools-default-containers-z' does not exist)
```

# Information to attach

 - [ ] Any relevant kernel output (`dmesg`)
 - [ ] Container log (`lxc info NAME --show-log`)
 - [ ] Container configuration (`lxc config show NAME --expanded`)
 - [ ] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log)
 - [ ] Output of the client with --debug
 - [ ] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)

### PR DESCRIPTION
**Pull request by tomponline**
Fixes #11675

### CODE REVIEW

File: shared/subprocess/proc.go [hunk 1]
```diff
@@ -41,6 +41,10 @@ type Process struct {
 }
 
 func (p *Process) hasApparmor() bool {
+	if os.Getenv("LXD_SECURITY_APPARMOR") == "false" {
```
Comment (monstermunchkin) [inline shared/subprocess/proc.go:44]: How about we use `shared.IsFalse`?

**Review by monstermunchkin — CHANGES_REQUESTED** [review]

File: shared/subprocess/proc.go (hunk 1 above)
Comment (tomponline) [reply shared/subprocess/proc.go:44]: I did think about that, but the existing usage of `os.Getenv("LXD_SECURITY_APPARMOR") ` compares to "false" so I didn't want to introduce a different type of logic here.

Comment (tomponline) [reply shared/subprocess/proc.go:44]: OK yeah, I wasn't sure if there was some historical context for that particularly strict check.

Comment (tomponline) [reply shared/subprocess/proc.go]: Fixed

**Pull request merged**